client/reducers/eventsplanning.ts

```diff
@@ -35,66 +36,59 @@ const eventsPlanningReducer = createReducer<IEventsPlanningState>(initialState,
 
     [INIT_STORE]: () => (initialState),
 
-    [EVENTS_PLANNING.ACTIONS.SET_EVENTS_PLANNING_LIST]: (state, payload) => (
-        {
-            ...state,
-            eventsAndPlanningInList: (payload || []).map((e) => e._id),
-        }
-    ),
-    [EVENTS_PLANNING.ACTIONS.ADD_EVENTS_PLANNING_LIST]: (state, payload) => (
-        {
-            ...state,
-            eventsAndPlanningInList: uniq([
-                ...cloneDeep(state.eventsAndPlanningInList || []),
-                ...(payload || []).map((e) => e._id),
-            ]),
-        }
-    ),
-    [EVENTS_PLANNING.ACTIONS.CLEAR_EVENTS_PLANNING_LIST]: (state, payload) => (
-        {
-            ...state,
-            eventsAndPlanningInList: [],
-        }
-    ),
+    [EVENTS_PLANNING.ACTIONS.SET_EVENTS_PLANNING_LIST]: (state, payload) =>
+        produce(state, (draft) => {
+            draft.eventsAndPlanningInList = (payload || []).map((e) => e._id);
+        }),
+
+    [EVENTS_PLANNING.ACTIONS.ADD_EVENTS_PLANNING_LIST]: (state, payload) =>
+        produce(state, (draft) => {
+            const newIds = (payload || []).map((e) => e._id);
+            const existingIds = new Set(draft.eventsAndPlanningInList || []);
+
+            newIds.forEach((id) => {
+                if (!existingIds.has(id)) {
+                    draft.eventsAndPlanningInList.push(id);
+                }
+            });
+        }),
```

Comment:
Most relevant piece is this one. Before we were replacing the entire array and now we're just adding those that do not exist in the list yet.